mkchromecast/_arg_parsing.py

```diff
@@ -523,6 +523,8 @@ def raise_arg_type_error():
     """,
 )
 
+# TODO(xsdg): Probably best to replace this with --suppress-video.  Otherwise,
```

Comment:
I need help understanding the rationale behind this. Sending audio by default makes much more sense than video.

Reply:
The key to this comment is on the following line (line 527): we should be able to auto-detect most cases when we should be sending only audio.  So, if someone passes `-i filename.ogg`, it should be obvious that we should take an audio-only codepath.

Secondly, as things currently operate, using `-i filename.ogg --video` still works; it just shows a black screen instead of the generic "Default Media Receiver/Mkchromecast" spinner.  That means that even if we get the auto-detection wrong, things should still mostly work.

This is in large part because (3) `ffmpeg` is really smart.  So anytime we're using ffmpeg in the pipeline, it will generally do something reasonable by default.  We should take more advantage of that, instead of implementing complicated re-encoding logic directly in Mkchromecast.

And finally, if we get the auto-detection wrong, and the user actually cares or ffmpeg isn't doing the right thing, that's when they'd use the `--suppress-video` flag.  That should be rare.

\
The general goal here is to simplify the argument structure.  Every time we can reduce the usage of (or completely drop) an argument without decreasing functionality, that's almost always a big win for usability.

Reply:
Thank you for your reply. I think it's important I take the time to reply extensively so you see why I asked the question I asked. 

When I developed this abomination (wink), mkchromecast was designed for macOS, **just a systray app**. I was basically taking over another project that did the same with nodejs and the developer was shutting down, and I was broke and could not afford Spotify premium. That worked decently, and many people got interested and I started to receive requests to do something on Linux. Because I used Qt, the systray kind of worked, but the nodejs part did not. I found shipping a DMG with nodejs worked perfectly fine for macOS users (something that worked out of the box). However, for Linux, the nodejs part needed something else, and I started using `ffmpeg`. That's easy for Linux users; you can just use your package manager, and it installed easily, and Linux users had a high chance they already knew what `ffmpeg` was. macOS users? not so much. And then people asked for video support, and a function in the command line for this and for that, and it became all a mess.  Mkcrhomecast started drifting now to a command line tool due to all users' requests. That's a "grosso modo" why things got so messy with time, besides the fact this was my first project ever, and I am not a software developer. I just know about other stuff. 

> The key to this comment is on the following line (line 527): we should be able to auto-detect most cases when we should be sending only audio. So, if someone passes `-i filename.ogg`, it should be obvious that we should take an audio-only codepath.
> 

I understand this, but the `--video` flag must be there for a reason. I don't remember exactly why, but it was about transcoding video files explicitly avoided lots of problems. I'm not implying your rationale is wrong, but I am just trying to clarify that the flag was there for a reason (good or bad).  `ffmpeg` could not send video in a way that chromecast liked. It was so problematic that at some point I just let people send their own ffmpeg command. I hope you get to make it better. 

> Secondly, as things currently operate, using `-i filename.ogg --video` still works; it just shows a black screen instead of the generic "Default Media Receiver/Mkchromecast" spinner. That means that even if we get the auto-detection wrong, things should still mostly work.
> 

That's correct. But I also remember that the Chromecast will not play the video if you don't send the right video stuff.  It is likely you prove me wrong, and I am confident you will find a way to improve it. 

> This is in large part because (3) `ffmpeg` is really smart. So anytime we're using ffmpeg in the pipeline, it will generally do something reasonable by default. We should take more advantage of that, instead of implementing complicated re-encoding logic directly in Mkchromecast.
> 

That is correct, assuming that ideally, ffmpeg can take care of everything, which was not the case when I tested a couple of years ago. 

> And finally, if we get the auto-detection wrong, and the user actually cares or ffmpeg isn't doing the right thing, that's when they'd use the `--suppress-video` flag. That should be rare.
> 

If the flag is rare, why should it be there? That's what, at this point, I fail to see. In which circumstance do you pass the `--suppress-video` flag? when a video file is not correctly transcoded automatically by ffmpeg? I understand we are just going to use a command in the backend  like `ffmpeg -i somefile.extension someoutput` and we will hope for the best. Is that correct?

> The general goal here is to simplify the argument structure. Every time we can reduce the usage of (or completely drop) an argument without decreasing functionality, that's almost always a big win for usability.

100% with you on this. Usability is always a win. 

Best,